fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -0,0 +1,318 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	ptypes "github.com/golang/protobuf/ptypes"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	daemonservicePb "github.com/google/fleetspeak/fleetspeak/src/client/daemonservice/proto/fleetspeak_daemonservice"
+	clientConfigPb "github.com/google/fleetspeak/fleetspeak/src/client/generic/proto/fleetspeak_client_generic"
+	"github.com/google/fleetspeak/fleetspeak/src/common"
+	spb "github.com/google/fleetspeak/fleetspeak/src/common/proto/fleetspeak"
+	cpb "github.com/google/fleetspeak/fleetspeak/src/config/proto/fleetspeak_config"
+	fcpb "github.com/google/fleetspeak/fleetspeak/src/server/components/proto/fleetspeak_components"
+	grpcServicePb "github.com/google/fleetspeak/fleetspeak/src/server/grpcservice/proto/fleetspeak_grpcservice"
+	servicesPb "github.com/google/fleetspeak/fleetspeak/src/server/proto/fleetspeak_server"
+	"google.golang.org/grpc"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+	"time"
+)
+
+var (
+	mysqlDatabase = flag.String("mysql_database", "", "MySQL database name to use")
+	mysqlUsername = flag.String("mysql_username", "", "MySQL username to use")
+	mysqlPassword = flag.String("mysql_password", "", "MySQL password to use")
+)
+
+type componentPids struct {
+	serverPid  int
```

Comment:
After another look on this PR, these should actually not be PIDs, but exec.Cmd instances.

This would also simplify the killing, since killing would mean:

cmd.Process.Signal()
cmd.Process.Wait()

and startProcess would look like:

func startProcess(...) ... {
 // setup
 // start goroutine to copy stdout
 // start goroutine to copy stderr
}